### PR TITLE
add apollo-tracing support

### LINKED ISSUE #219: Support of Apollo Tracing?
Hi, are there any plans to support Apollo Tracing? Not sure if it would belong in this library, but just wanted to check.

### PR DESCRIPTION
**Pull request by vvakame**
closes #219
https://github.com/apollographql/apollo-tracing

I'll test this implementation on my app later.

I have:
 - [X] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [ ] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

File: handler/graphql.go [hunk 1]
```diff
@@ -271,35 +298,58 @@ func GraphQL(exec graphql.ExecutableSchema, options ...Option) http.HandlerFunc
...
 		}
 		if doc == nil {
-			var qErr gqlerror.List
-			doc, qErr = gqlparser.LoadQuery(exec.Schema(), reqParams.Query)
-			if len(qErr) > 0 {
-				sendError(w, http.StatusUnprocessableEntity, qErr...)
+			var gqlErr *gqlerror.Error
+			var listErr gqlerror.List
+
+			ctx = cfg.tracer.StartOperationParsing(ctx)
```
Comment (vektah) [inline handler/graphql.go]: its probably time to break this into a separate function so the ends can be deferred. This currently looks like a recipe for bugs soup.

**Comment by vvakame** [comment]
Looks fine 🤔 
tracing_example

**Comment by GarethSharpe** [comment]
Any documentation on how to implement this?

**Comment by mathewbyrne** [comment]
I don't believe so, the contrib package is in the https://github.com/99designs/gqlgen-contrib repo, and could definitely do with a README at least.  Feel free to open an issue there.

**Comment by vniche** [comment]
for new-comers, here is a great example: https://github.com/99designs/gqlgen-contrib/blob/master/gqlapollotracing/example_test.go

**Pull request merged**